crates/chain/src/keychain/txout_index.rs

```diff
@@ -463,16 +632,39 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     /// Get the last derivation index that is revealed for each keychain.
     ///
     /// Keychains with no revealed indices will not be included in the returned [`BTreeMap`].
-    pub fn last_revealed_indices(&self) -> &BTreeMap<K, u32> {
-        &self.last_revealed
+    pub fn last_revealed_indices(&self) -> BTreeMap<K, u32> {
+        self.last_revealed
+            .iter()
+            .map(|(descriptor_id, index)| {
+                (
+                    self.keychain_of_descriptor_id(descriptor_id)
+                        .expect("Must be here")
```

Comment:
IF there is no corresponding keychain, just don't return it.

Reply:
I created #1348 to take a stab at returning a `Result<ChangeSet<K>, AddKeychainError>`.